web-ts/docs/technical-documentation.md

```diff
@@ -0,0 +1,94 @@
+# Technical Documentation
+
+## Contents
+
+- [Technology Stack](#technology-stack)
+- [Testing](#testing)
+- [Hidden Cities](#hidden-cities)
+- [Code Quality](#code-quality)
+- [Backend](#backend)
+
+## Technology Stack
+
+### Bundler
+
+[Webpack](https://webpack.github.io/) is used to compile and bundle the app.
+
+### Application state
+
+[Redux](http://redux.js.org/) is used for the global application state.
+The data which is received through the restful api of the CMS is "cached" and stored in this state container.
+
+## Testing
+
+### Unit Tests
+
+[Jest](https://facebook.github.io/jest/) is used for testing.
+
+**For testing of React components the [testing-library](https://testing-library.com) should be used whenever possible.**
+Testing the application components should be done in the way the user would use it as stated in its [guiding principles](https://testing-library.com/docs/guiding-principles).
+
+### E2E Tests
+
+Refer to the corresponding [documentation](../../docs/e2e-tests.md).
+You can use [Browserstack](https://www.browserstack.com) for testing cross-browser compatibility.
+
+## Code Quality
+
+### Formatting
+
+We use [prettier](https://prettier.io) to format code.
+Run `yarn workspace <workspace> prettier --check .` to show formatting problems or `yarn workspace <workspace> prettier --write .` to fix them.
+
+### Linting
+
+Linting is responsible to embrace best practice coding style.
+We use [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) for TypeScript and [stylelint](https://stylelint.io/) for CSS.
+
+You can run the linter by calling `yarn lint`.
+
+### Folder Structure
+
+First level structure of the `src-Folder` is technical specific and includes routes, reusable components, hooks, contexts and services.
+Inside the `routes-Folder`, subfolders are domain specific and can have - if necessary - also technical separation.
+File naming should be PascalCase, inside routes-Folder file names should be prefixed by the route name.
+Router entry points should have `Page` as a suffix, services should have `Service` suffix.
+
+```
+src
+└───routes
+│   └───SomeRoute
+│       │   SomeRoutePage.tsx
+│       └───component
+│       │   │ SomeRouteItem.tsx
+│       │   │ SomeRouteList.tsx
+│       └───service
+│           │ SomeRouteSpecificService.ts
+└───components
+│   │   Button.tsx
+│   │   Icon.tsx
+└───hooks
+└───context
+└───services
+    └───Format
+        │   DateFormatterService.ts
```

Comment:
Not sure if i understand you correctly, but it would be in separate modules. So a services used only inside on route, would be in route/my-route/service/route-specific-service.ts
an the shared services can also be separated inside the servies-folder (like Format-Subfolder as in the example)

Reply:
Currently we have the `modules` directory containing things like `i18n`, `layout` or `endpoint`. It looks like this is not the case in this folder structure anymore but instead its just sorted according to the technical differentiation into hooks, services,... And I am not sure if that is good for a clear structure/finding the right things.

Reply:
Let's discuss this later today or tomorrow in the team call :)

Reply:
Will discuss this in app call